server/watch_volfile_changes.py

```diff
@@ -0,0 +1,92 @@
+import os
+import time
+import json
+import logging
+import signal
+import serverutils
+from kadalulib import (CommandException, execute, logf,
+                       logging_setup)
+from glusterfsd import (create_brick_volfile, create_client_volfile)
+from shd import create_shd_volfile
+
+VOLINFO_DIR = "/var/lib/gluster"
+VOLFILES_DIR = "/var/lib/kadalu/volfiles"
+
+brick_device = os.environ.get("BRICK_DEVICE", None)
+brick_path = os.environ["BRICK_PATH"]
+
+volume_id = os.environ["VOLUME_ID"]
+brick_path_name = brick_path.strip("/").replace("/", "-")
+volname = os.environ["VOLUME"]
+nodename = os.environ["HOSTNAME"]
+
+volfile_id = "%s.%s.%s" % (volname, nodename, brick_path_name)
+
+storage_unit_volfile_path = os.path.join(VOLFILES_DIR, "%s.vol" % volfile_id)
+client_volfile_path = os.path.join(VOLFILES_DIR, "%s.vol" % volname)
+shd_volfile_path = os.path.join(VOLFILES_DIR, "glustershd.vol")
+
+data = {}
+info_file_path = os.path.join(VOLINFO_DIR, "%s.info" % volname)
+
+def send_sighup():
+    """ Send SIGHUP to Brick & SHD Process notifing change in volfiles """
+
+    try:
+        out, _, _ = execute("pgrep", "-f", "-a", "glusterfsd|shd")
+        for line in out.split("\n"):
+            pid = line.split(" ", 1)[0]
+            os.kill(int(pid), signal.SIGHUP)
+    except CommandException as err:
+        logging.error(logf(
+            "Failed to get pid",
+            error=err
+        ))
+    except ProcessLookupError as err:
```

Comment:
Move this exception handling inside for loop and include `line` in the log details along with the `error=err`.

Reply:
Updated